models/preresnet.lua

```diff
@@ -32,7 +32,6 @@ local function createModel(opt)
          -- 1x1 convolution
          return nn.Sequential()
             :add(Convolution(nInputPlane, nOutputPlane, 1, 1, stride, stride))
-            :add(SBatchNorm(nOutputPlane))
```

Comment:
In the paper the authors said
> For the bottleneck ResNets, when reducing the feature map size we use projection shortcuts [1] for increasing dimensions, and when pre-activation is used, these projection shortcuts are also with pre-activation.

And it is also implemented in their code [here](https://github.com/KaimingHe/resnet-1k-layers/blob/master/resnet-pre-act.lua#L83).

Thus this batch norm is actually placed before the Convolution layer, and the whole unit belongs to the `both_preact` case.